KubeArmor/kvmAgent/kvmAgent.go

```diff
@@ -0,0 +1,120 @@
+package kvmAgent
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"io"
+	"log"
+	"os"
+
+	tp "github.com/kubearmor/KubeArmor/KubeArmor/types"
+	pb "github.com/kubearmor/KubeArmor/protobuf"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+)
+
+// Variables
+var (
+	client           pb.KVMClient
+	identity         string
+	grpcClientConn   grpc.ClientConn
+	UpdateHostPolicy func(tp.K8sKubeArmorHostPolicyEvent)
+)
+
+func getGrpcConnAddress() string {
+	return (os.Getenv("CLUSTER_IP") + ":" + os.Getenv("CLUSTER_PORT"))
+}
+
+func enforcePolicy(policyBytes []byte) error {
+	err := *new(error)
+	err = nil
+	policyEvent := tp.K8sKubeArmorHostPolicyEvent{}
+
+	err = json.Unmarshal(policyBytes, &policyEvent)
+	if err != nil {
+		log.Print("Unmarshal error => ", err)
+	} else {
+		UpdateHostPolicy(policyEvent)
+	}
+
+	return err
+}
+
+func connectToKVMService() error {
+	err := *new(error)
+	var status int32
+	err = nil
+	status = 0
+
+	md := metadata.New(map[string]string{"identity": identity})
+	ctx := metadata.NewOutgoingContext(context.Background(), md)
+
+	stream, err := client.SendPolicy(ctx)
+	if err != nil {
+		log.Print("Failed to stream")
+		grpcClientConn.Close()
```

Comment:
Use defer grpcClientConn.Close() in the beginning of the function. The best would be to use defer in the function that created this connection.